File: solutions/lua/perfect-numbers/1/perfect-numbers.lua
local function aliquot_sum(n)
  s=0
  for i = 1, n-1, 1
  do
    if n % i == 0
    then
      s=s+i
    end
  end
  return s
end

local function classify(n)
  if aliquot_sum(n) == n 
  then 
    return 'perfect'
  elseif aliquot_sum(n) > n
  then 
    return 'abundant'
  else
    return 'deficient'
  end
end

return {
  aliquot_sum = aliquot_sum,
  classify = classify
}

File: solutions/javascript/perfect-numbers/1/perfect-numbers.js

export const classify = (num) => {
  if (num <= 0)
    throw new Error('Classification is only possible for natural numbers.');
  var fact = factors(num);
  if (num != 1 && sum(fact) === num) return "perfect"
  if (sum(fact) > num) return "abundant"
  if (num === 1 || sum(fact) < num) return "deficient"
};
function factors(num){
  var ret = new Array();
  ret.push(1);
  for (var i=2;i<num;i++) {
    if (num % i == 0) ret.push(i);
  }
  return ret;
}
function sum(factors){
  var ret = 0;
  factors.forEach(fact => ret += fact);
  return ret;
}